src/Microsoft.AspNetCore.SignalR.Client/InvocationRequest.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    internal class InvocationRequest : IDisposable
```

Comment:
Should we have two requests implementing the same interface? We fork the code basically in each function based on the _streaming flag...